hardware/smart_utils.py

```diff
@@ -19,6 +19,13 @@
 from hardware.detect_utils import which
 
 
+def _parse_line(line):
+    line = line.strip()
+    if isinstance(line, six.binary_type):
+        line = line.decode(errors='ignore')
```

Comment:
ok, you corrected it already, cool!
If you can move it to detect_utils, call it from there and add the unit test, it would be perfect :)